### PR TITLE
Add fully functional state facility frontend

### PR DESCRIPTION
**Pull request by WryanCartie**
Add fully functional state facility frontend to display facilities of a state

### CODE REVIEW

File: src/renderers/stateRenderer.js [hunk 1]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
 require('bootstrap');
 
 // START UTILITY FUNCTIONS
+
+function sortTable(n,tableId) {
+    let table, rows, switching, i, x, y, shouldSwitch, dir, switchcount = 0;
+    table = document.getElementById(tableId)
+    console.log(table);
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:12]: Do we need to log this?

File: src/renderers/stateRenderer.js [hunk 2]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
...
+
+    switching = true;
+    // Set the sorting direction to ascending:
+    dir = "asc";
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:16]: Do we need that comment? How about renaming the variable to `sortingDirection` or something like that?

File: src/renderers/stateRenderer.js [hunk 3]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
...
+    /* Make a loop that will continue until
+    no switching has been done: */
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:18]: Remove comment

File: src/renderers/stateRenderer.js [hunk 4]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
...
+    while (switching) {
+      // Start by saying: no switching is done:
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:20]: Remove comment

File: src/renderers/stateRenderer.js [hunk 5]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
...
+      switching = false;
+      rows = table.rows;
+   
+      /* Loop through all table rows (except the
+      first, which contains table headers): */
+      for (i = 1; i < (rows.length - 1); i++) {
+        // Start by saying there should be no switching:
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:27]: Remove comment

File: src/renderers/stateRenderer.js [hunk 6]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
...
+        shouldSwitch = false;
+        /* Get the two elements you want to compare,
+        one from current row and one from the next: */
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:30]: Remove comment

File: src/renderers/stateRenderer.js (hunk 2 above)
Comment (Phobez) [reply src/renderers/stateRenderer.js:16]: And extract it into a `const`

File: src/renderers/stateRenderer.js [hunk 7]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
...
+        x = rows[i].getElementsByTagName("TD")[n];
+        y = rows[i + 1].getElementsByTagName("TD")[n];
+        /* Check if the two rows should switch place,
+        based on the direction, asc or desc: */
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:34]: Remove the comment

File: src/renderers/stateRenderer.js [hunk 8]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
...
+        if (dir == "asc") {
+          if (x.innerHTML.toLowerCase() > y.innerHTML.toLowerCase()) {
+            // If so, mark as a switch and break the loop:
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:37]: Remove comment

File: src/renderers/stateRenderer.js [hunk 9]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
...
+            shouldSwitch = true;
+            break;
+          }
+        } else if (dir == "desc") {
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:41]: Extract this into a `const` too

File: src/renderers/stateRenderer.js [hunk 10]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
...
+          if (x.innerHTML.toLowerCase() < y.innerHTML.toLowerCase()) {
+            // If so, mark as a switch and break the loop:
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:43]: Remove comment

File: src/renderers/stateRenderer.js [hunk 11]
```diff
@@ -5,6 +5,66 @@ const fs = require('fs');
...
+            break;
+          }
+        }
+      }
+      if (shouldSwitch) {
+        /* If a switch has been marked, make the switch
+        and mark that a switch has been done: */
+        rows[i].parentNode.insertBefore(rows[i + 1], rows[i]);
+        switching = true;
+        // Each time a switch is done, increase this count by 1:
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:54]: Remove comment

File: src/renderers/stateRenderer.js [hunk 12]
```diff
@@ -157,6 +220,65 @@ function getRegions() {
...
+                }
+                let regionTemplate = `
+                    <tr>
+                        <td scope="col">${facility.regionName}</a></td>
+                        <td scope="col" >${facility.facilityName}</td>
+                        <td scope="col" >${facilityStatus}</td>
+                    </tr>`;
+
+                $('#StateFacilities').append(regionTemplate);
+            });
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:266]: Can we extract this into a function by itself?

**Review by Phobez — CHANGES_REQUESTED** [review]

File: src/renderers/stateRenderer.js (hunk 1 above)
Comment (WryanCartie) [reply src/renderers/stateRenderer.js:12]: Removed

File: src/renderers/stateRenderer.js (hunk 12 above)
Comment (WryanCartie) [reply src/renderers/stateRenderer.js:266]: There's actually two in this case, one for the table and one that injects the data for the table

File: src/renderers/stateRenderer.js (hunk 11 above)
Comment (WryanCartie) [reply src/renderers/stateRenderer.js:54]: So should i just remove all comments?
'\
'

File: src/renderers/stateRenderer.js (hunk 12 above)
Comment (Phobez) [reply src/renderers/stateRenderer.js:261]: The entire thing actually, the thing inside the latest `{}`, lines 252-266

File: src/renderers/stateRenderer.js [hunk 13]
```diff
@@ -157,6 +235,48 @@ function getRegions() {
     });
 }
 
+function getFacilities() {
+    ipcRenderer.send("Facility:getFacilitiesByState", parseInt(getProcessArgObj()));
+    ipcRenderer.once("Facility:getFacilitiesByStateOK", (e, res) => {
+        $('#listOfFacilities').empty();
+        let table = `<table id="state-facility" class="table table-striped table-bordered table-sm" cellspacing="0" width="100%">
```
Comment (Phobez) [inline src/renderers/stateRenderer.js:242]: Let's extract this like we talked about

**Review by Phobez — CHANGES_REQUESTED** [review]

File: src/renderers/stateRenderer.js (hunk 13 above)
Comment (WryanCartie) [reply src/renderers/stateRenderer.js:242]: You want me to make another function. For table creation?

**Review by Phobez — APPROVED** [review]

**Pull request merged**